opm/simulators/flow/BlackoilModelEbosNldd.hpp

```diff
@@ -0,0 +1,801 @@
+            const std::ios::fmtflags oflags = ss.setf(std::ios::scientific);
+            ss << std::setw(4) << iteration;
+            for (int compIdx = 0; compIdx < numComp; ++compIdx) {
+                ss << std::setw(11) << mass_balance_residual[compIdx];
+            }
+            for (int compIdx = 0; compIdx < numComp; ++compIdx) {
+                ss << std::setw(11) << CNV[compIdx];
+            }
+            ss.precision(oprec);
+            ss.flags(oflags);
+            OpmLog::debug(ss.str());
+        }
+
+        return report;
+    }
+
+    ConvergenceReport getDomainConvergence(const Domain& domain,
+                                           const SimulatorTimerInterface& timer,
+                                           const int iteration,
+                                           std::vector<double>& residual_norms)
+    {
+        std::vector<Scalar> B_avg(numEq, 0.0);
+        auto report = this->getDomainReservoirConvergence(timer.simulationTimeElapsed(),
+                                                          timer.currentStepLength(),
+                                                          iteration,
+                                                          domain,
+                                                          B_avg,
+                                                          residual_norms);
+        report += model_.wellModel().getDomainWellConvergence(domain, B_avg);
+        return report;
+    }
+
+    //! \brief Returns subdomain ordered according to method and ordering measure.
+    std::vector<int> getSubdomainOrder()
+    {
+        const auto& ebosSimulator = model_.ebosSimulator();
+        const auto& solution = ebosSimulator.model().solution(0);
+
+        std::vector<int> domain_order(domains_.size());
+        switch (model_.param().local_solve_approach_) {
+        case DomainSolveApproach::GaussSeidel: {
+            switch (model_.param().local_domain_ordering_) {
+            case DomainOrderingMeasure::AveragePressure: {
+                // Use average pressures to order domains.
+                std::vector<std::pair<double, int>> avgpress_per_domain(domains_.size());
+                for (const auto& domain : domains_) {
+                    double press_sum = 0.0;
+                    for (const int c : domain.cells) {
+                        press_sum += solution[c][Indices::pressureSwitchIdx];
+                    }
+                    const double avgpress = press_sum / domain.cells.size();
+                    avgpress_per_domain[domain.index] = std::make_pair(avgpress, domain.index);
+                }
+                // Lexicographical sort by pressure, then index.
+                std::sort(avgpress_per_domain.begin(), avgpress_per_domain.end());
+                // Reverse since we want high-pressure regions solved first.
+                std::reverse(avgpress_per_domain.begin(), avgpress_per_domain.end());
```

Comment:
You strictly speaking need `<algorithm>` for `std::sort()` and `std::reverse()`.

Reply:
yeah, actually noticed and fixed locally but forgot to push.